src/Raven.Server/Smuggler/Documents/DatabaseDestination.cs

```diff
@@ -1461,6 +1462,11 @@ protected override long ExecuteCmd(DocumentsOperationContext context)
                         {
                             _database.DocumentsStorage.RevisionsStorage.Put(context, id, document.Data, document.Flags,
                                 document.NonPersistentFlags, document.ChangeVector, document.LastModified.Ticks);
+
+                            if (addHasRevisionsFlag.Contains(id) == false && document.Flags.Contain(DocumentFlags.HasRevisions))
```

Comment:
This isn't the document but the revision.
You need to check if the actual document is missing the `Revisions` flag.

Reply:
done